src/app/embed/adapters/merge.adapter.ts

```diff
@@ -0,0 +1,38 @@
+import { Http } from '@angular/http';
+import { EMBED_AUDIO_URL_PARAM, EMBED_TITLE_PARAM, EMBED_SUBTITLE_PARAM,
+  EMBED_SUBSCRIBE_URL_PARAM, EMBED_SUBSCRIBE_TARGET, EMBED_IMAGE_URL_PARAM, EMBED_FEED_ID_PARAM, EMBED_EPISODE_GUID_PARAM } from './../embed.constants';
+import { QSDAdapter } from './qsd.adapter'
+import { FeedAdapter } from './feed.adapter'
+
+import { Observable } from 'rxjs/Observable';
+
+export class MergeAdapter {
+  private QSDAdapter: QSDAdapter
```

Comment:
Rather than automatically constructing and then keeping a set number of arguments, you should be accepting an array or varargs of "Adapter" arguments, where adapter is an interface that the different adapters implement. Probably as simple as`getParams(): Observable<AdapterProperties>` where `AdapterProperties` is the interface I described earlier.